core/src/main/java/com/novoda/noplayer/PlayerAudioTrack.java

```diff
@@ -60,12 +78,13 @@ public boolean equals(Object o) {
             return false;
         }
         return mimeType != null ? mimeType.equals(that.mimeType) : that.mimeType == null;
-
     }
 
     @Override
     public int hashCode() {
-        int result = trackId != null ? trackId.hashCode() : 0;
+        int result = groupIndex;
+        result = 31 * result + trackIndex;
```

Comment:
No. `groupIndex` and `trackIndex` cannot be null because they are primitives. Only the `trackId`, `language` and `mimetype` can be null.